src/document/Document.js

```diff
@@ -675,10 +675,12 @@ define(function (require, exports, module) {
     
     /**
      * Updates the language to match the current mapping given by LanguageManager
+     * 
+     * @param {!string} rawText  Text content of the file.
      */
-    Document.prototype._updateLanguage = function () {
+    Document.prototype._updateLanguage = function (rawText) {
```

Comment:
This is called from other places that won't pass `rawText` (e.g. `_notifyFilePathChanged()`, `DocumentManager._handleLanguageAdded()`, etc.).  Will the behavior be correct then?  It safer & simpler to remove the arg and just invoke `this.getText()` instead.